drivers/usb/udc/udc_common.c

```diff
@@ -46,6 +46,31 @@ struct udc_ep_config *udc_get_ep_cfg(const struct device *dev, const uint8_t ep)
 	return data->ep_lut[USB_EP_LUT_IDX(ep)];
 }
 
+bool udc_ep_is_busy(const struct device *dev, const uint8_t ep)
+{
+	struct udc_ep_config *ep_cfg;
+
+	ep_cfg = udc_get_ep_cfg(dev, ep);
+	if (unlikely(ep_cfg == NULL)) {
+		__ASSERT(false, "ep 0x%02x is not available", ep);
+		return true;
+	}
```

Comment:
I don't like such concepts. I'd prefer to replace it with `__ASSERT(ep_cfg != NULL, "ep 0x%02x is not available", ep);`. The thing is, if this is called with invalid ep, then the code is already known to be wrong and continuting execution is going no good. It is better if it just crashes with NULL pointer dereference if the code is wrong *and* asserts are disabled instead of masking the issue (which essentially makes debugging harder and allows bugs to go unnoticed for long time - or even worse, worked around in higher layers).

Reply:
>It is better if it just crashes with NULL pointer dereference if the code is wrong and asserts are disabled instead of masking the issue...

That actually makes perfect sense in this case. I will change it.